test/compress/collapse_vars.js

```diff
@@ -2256,3 +2256,31 @@ issue_2187_3: {
     }
     expect_stdout: "1"
 }
+
+issue_2203_1: {
+    options = {
+        collapse_vars: true,
+        unused: true,
+    }
+    input: {
+        console.log({
+            a: 42,
+            b: function() {
+                return function(c) {
+                    return c.a;
+                }(this);
```

Comment:
Umm, wait a second - you do realise this PR is for `master`, right? :sweat_smile: